Data/Vector.hs

```diff
@@ -1474,6 +1475,23 @@ ifoldr' :: (Int -> a -> b -> b) -> b -> Vector a -> b
 {-# INLINE ifoldr' #-}
 ifoldr' = G.ifoldr'
 
+-- | @since NEXT
+-- /O(n)/ Map each element of the structure to a monoid, and combine
+-- the results. This function is implemented in terms of 'foldr' in
+-- the same way as default implementation in 'Foldable' type class.
```

Comment:
We could avoid refering to implementation details saying "Map each element of the structure to a monoid, and combine results {lazily,strictly}, similar to `Data.Foldable.{foldMap,foldMap'}`".

Reply:
At first I removed foldl/foldr mentions as implementation detial. But then decided to restore it. Whether function will fuse or not depends whether foldl/foldr is used. It's unfortunate that implementation details leak like that